File: Back/src/ProEventos.Domain/RedeSocial.cs
namespace ProEventos.Domain
{
    public class RedeSocial
    {
        public int Id { get; set; }
        public string Nome { get; set; }
        public string URL { get; set; }

        // Chave estrangeira
        public int? EventoId { get; set; }
        public Evento Evento { get; set; }
        public int? PalestranteId { get; set; }
        public Palestrante Palestrante { get; set; }
    }
}

File: Front/ProEventos-App/src/app/models/RedeSocial.ts

import { Evento } from './Evento';
import { Palestrante } from './Palestrante';

export interface RedeSocial {

  id: number;
  nome: string;
  URL: string;

  // Chave estrangeira
  eventoId: number;
  evento: Evento;
  palestranteId?: number;
  palestrante: Palestrante;
}
